modP2pImpl/src/org/aion/p2p/impl/comm/NodeMgr.java

```diff
@@ -168,20 +165,17 @@ public boolean isSeedIp(String _ip) {
     /** @param _n Node */
     @Override
     public void addTempNode(final INode _n) {
-        final ReentrantLock putLock = this.putLock;
+        tempNodesLock.lock();
```

Comment:
Previously, it used lockInterruptibly() instead of lock().  Should we also do lockInterruptibly here?

I'm not that familiar with the difference, but from the JavaDoc, it seems when the thread blocks on lock(), it can't be interrupted until the lock is acquired.  Not sure if this will do something strange if the kernel shuts down while this thread is blocked on lock()

Reply:
yap, the lockInterruptibly it a little bit better in this scenario.